samples/detect.v2beta1.js

```diff
@@ -0,0 +1,822 @@
+    projectPath: projectId,
+  });
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  //const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+
+  var formattedParent = client.projectPath(projectId);
+
+  client
+    .listKnowledgeBases({parent: formattedParent})
+    .then(responses => {
+      var resources = responses[0];
+      resources.forEach(r => {
+        console.log(`  Display name is : ${r.displayName}`);
+        console.log(`  Knowledge base is : ${r.name}`);
+      });
+    })
+    .catch(err => {
+      console.error(err);
+    });
+  // [END dialogflow_list_knowledge_base]
+}
+
+function deleteKnowledgeBase(projectId, knowledgeBaseFullName) {
+  // [START dialogflow_delete_knowledge_base]
+  // Instantiate a DialogFlow client.
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow KnowledgeBasesClient.
+  var client = new dialogflow.KnowledgeBasesClient();
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  //const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+  //const knowledgeBaseFullName = `the full path of your knowledge base, e.g my-Gcloud-project/myKnowledgeBase`;
+
+  const fullName = knowledgeBaseFullName;
+
+  client
+    .deleteKnowledgeBase({
+      name: fullName,
+    })
+    .then(responses => {
+      const result = responses[0];
+      console.log(`Name: ${result.name}`);
+      console.log(`displayName: ${result.displayName}`);
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_delete_knowledge_base]
+}
+
+function createDocument(
+  projectId,
+  knowledgeBaseFullName,
+  documentPath,
+  documentName
+) {
+  // [START dialogflow_create_document]
+  // Imports the Dialogflow client library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow Documents client.
+  var client = new dialogflow.DocumentsClient({
+    projectId: projectId,
+  });
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  //const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+  //const knowledgeBaseFullName = `the full path of your knowledge base, e.g my-Gcloud-project/myKnowledgeBase`;
+  //const documentPath = `path of the document you'd like to add, e.g. https://dialogflow.com/docs/knowledge-connectors`;
+  //const documentName = `displayed name of your document in knowledge base, e.g. myDoc`;
+
+  var request = {
+    parent: knowledgeBaseFullName,
+    document: {
+      knowledgeTypes: [`FAQ`],
+      displayName: documentName,
+      contentUri: documentPath,
+      source: `contentUri`,
+      mimeType: `text/html`,
+    },
+  };
+
+  client
+    .createDocument(request)
+    .then(console.log(`Document created`))
```

Comment:
Isn't this a Long Running Operation (LRO)? In which case, when the request is returned, the document is not created. Unless the `.then(` handles this or something?

I'd TAL at our other LRO samples, eg. for Speech or Video...
 - https://cloud.google.com/speech-to-text/docs/async-recognize
 - https://cloud.google.com/video-intelligence/docs/analyze-labels

<img width="731" alt="screen shot 2018-08-02 at 3 56 30 pm" src="https://user-images.githubusercontent.com/280/43615485-ad6fff64-966c-11e8-95bd-36c8b6142898.png">

Reply:
CreateDocument definition (returns an Operation, ie LRO)
https://cloud.google.com/dialogflow-enterprise/docs/reference/rpc/google.cloud.dialogflow.v2beta1#google.cloud.dialogflow.v2beta1.Documents.CreateDocument